interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/ParameterEvaluator.scala

```diff
@@ -3,13 +3,21 @@ package pl.touk.nussknacker.engine.compile.nodecompilation
 import pl.touk.nussknacker.engine.api.context.transformation._
 import pl.touk.nussknacker.engine.api.definition.{AdditionalVariableWithFixedValue, Parameter => ParameterDef}
 import pl.touk.nussknacker.engine.api.expression.{TypedExpression, TypedExpressionMap}
-import pl.touk.nussknacker.engine.api.{Context, MetaData, NodeId}
+import pl.touk.nussknacker.engine.api.{Context, LazyParameter, MetaData, NodeId, ProcessListener}
 import pl.touk.nussknacker.engine.compiledgraph.{CompiledParameter, TypedParameter}
 import pl.touk.nussknacker.engine.expression.ExpressionEvaluator
 import pl.touk.nussknacker.engine.graph
 import pl.touk.nussknacker.engine.util.Implicits._
+import pl.touk.nussknacker.engine.variables.GlobalVariablesPreparer
 
-class ParameterEvaluator(expressionEvaluator: ExpressionEvaluator) {
+class ParameterEvaluator(
```

Comment:
`Parameter` need to be typed because evaluation requires that. Depending on the `TypingResult`, we do some implicit conversions (aligments) on the end. So `Typed` it's rather a detail instead of a domain concept. But I don't have a strong opinion in this area.